### ISSUE TITLE
Install on Windows fails - Prior to new release it installed without error

### ISSUE THREAD

**Issue by bb-ceccles** [garris/BackstopJS#499]
I have the old release installed globally and it is working fine with the phantom engine.

Myself and others have been testing the new beta and had this same issue last week. I have seen that the new version has been releases today so tried again.

I am still getting an error when the install is attempting to install the sharp package.

The error it is throwing is related to missing build tools but both the v110 and v140 tools are installed.

This error is thrown whether you install globally through npm or locally

npm install -g backstopjs

## Error message

```
if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "" rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V110\Microsoft.Cpp.Platform.targets(44,5): error MSB8020: The builds
tools for v140 (Platform Toolset = 'v140') cannot be found. To build using the v140 build tools, either click the Proje
ct menu or right-click the solution, and then select "Update VC++ Projects...". Install v140 to build using the v140 bu
ild tools. [C:\Users\ceccles\AppData\Roaming\npm\node_modules\backstopjs\node_modules\sharp\build\libvips-cpp.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `msbuild` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:258:23)
gyp ERR! stack     at emitTwo (events.js:125:13)
gyp ERR! stack     at ChildProcess.emit (events.js:213:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Windows_NT 10.0.15063
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\\.....MY_LOCAL.......\AppData\Roaming\npm\node_modules\backstopjs\node_modules\sharp
gyp ERR! node -v v8.4.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! sharp@0.18.2 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the sharp@0.18.2 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

**Comment by garris** [comment]
@bb-ceccles Thanks for posting!  I think @patriciamarchand had a very similar issue.  It required two fixes...

https://github.com/garris/BackstopJS/issues/479#issuecomment-319745257

Please let me know if this addresses your issue.   I should probably add this to the readme for windows users.  

FWIW:  sharp won't be required after the next release of chrome.

**Comment by garris** [comment]
Update: @dotneet has updated chromy and removed the sharp dependency.  I will post back to this thread when I have update backstopjs with the new build.   Thanks for your patience.

**Comment by bb-ceccles** [comment]
Thanks for the update, I have tried to install the "Windows Software Development Kit (SDK) for Windows 8.1"

This didn't help but would be great if sharp is removed as a dependency as that is where the install is tripping up.

Thanks again for the quick response.  We have backstop running on mac and it is amazing and a big help especially with the new Chrome integration so thanks.

**Comment by garris** [comment]
Great to hear!   I just pushed up a new version with the sharp library removed. Please try it again and let me know if it's working for you again.
```
npm install -g backstopjs
```
Thanks!

**Comment by bb-ceccles** [comment]
Hi @garris 

Thanks for the speedy response again. 

The install works correctly and did not throw any errors. On initial run it was throwing errors reporting the chromy expected chrome version 61 (current live version is 60 sitll)

I have had to install chrome canary (v61) but this has then resolved the issues and allowed me to run the tests.

The tests are not passing due to minimal change differences in the comparisons. I am not sure if this is a chrome version difference or the chromy plugin or just difference in font rendering between mac and PC. Website is still in development :-)

See fail example.

![image](https://user-images.githubusercontent.com/5575674/29573356-ad914a50-8756-11e7-8d8b-15505d7b0cbd.png)

I will look in to the issues and try and report back if there is anything found related to the library. I have a feeling it is just font related.

**Comment by patriciamarchand** [comment]
@bb-ceccles I had the same problem due to minimal change differences in the comparisons. I think is browser rendering related #487.

I will try also to install again Backstop on a new machine as soon as I can.

**Comment by garris** [comment]
FYI:  I have filed this...  https://github.com/OnetapInc/chromy/issues/61

**Comment by EricBou** [comment]
@garris 

I work with Patricia and I just discovered the cause of his error

When I look at the list of versions in github, I find version 2.7.3 to be the most recent. However, NPM finds versions 2.7.5 and 2.8.4. Is it voluntary or is an error.  Just run "npm show backstopjs@* version"

In addition, in version 2.8.4, the dependency chromy has been added. This causes problems with the installation of sharp on several computers that I use. Is that chromy is mandatory with this version or it is going to be a new dependency only with version 3.x.

Currently I have freeze backstop to version 2.7.5 to fix the problem.

Regards

Eric
